library/cpp/protocol/TBinaryProtocol.tcc

```diff
@@ -665,20 +657,12 @@ int32_t TBinaryProtocolT<Transport_>::readIPADDR(
   }
   if (version == AF_INET) {
     boost::asio::ip::address_v4::bytes_type ipv4;
-#ifdef _WINDOWS
     this->trans_->readAll(ipv4.data(), 4);
-#else
-    this->trans_->readAll(ipv4.c_array(), 4);
-#endif
     ipaddress = boost::asio::ip::address_v4(ipv4);
     return ret+4;
   } else if (version == AF_INET6) {
     boost::asio::ip::address_v6::bytes_type ipv6;
-#ifdef _WINDOWS
     this->trans_->readAll(ipv6.data(), 16);
```

Comment:
`readAll` needs pointer to non-const data. `boost:array::data()` returns pointer to const data. Please check if it works with `boost::array`.